app/src/androidTest/java/ch/epfl/sdp/lobby/game/GameLobbyActivityTest.kt

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.sdp.lobby.game
+
+import androidx.recyclerview.widget.RecyclerView
+import androidx.test.core.app.launchActivity
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.swipeDown
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import ch.epfl.sdp.R
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class GameLobbyActivityTest {
+
+    @Test
+    fun activityDoesNotCrash() {
+        launchActivity<GameLobbyActivity>()
+        onView(ViewMatchers.withId(R.id.swipe_container)).perform(swipeDown())
+    }
+
+    @Test
+    fun swipeDownRefreshesData() {
+        val scenario = launchActivity<GameLobbyActivity>()
+
+        var adapter = GameLobbyAdapter(Collections.emptyList(),0,0)
+        scenario.onActivity { a ->
+            adapter = a.findViewById<RecyclerView>(R.id.player_list).adapter as GameLobbyAdapter
+        }
+
+        onView(ViewMatchers.withId(R.id.faction_selection)).perform(click())
+        onView(ViewMatchers.withId(R.id.player_list)).perform(swipeDown())
+
+        var newAdapter = GameLobbyAdapter(Collections.emptyList(),0,0)
+        scenario.onActivity { a ->
+            newAdapter = a.findViewById<RecyclerView>(R.id.player_list).adapter as GameLobbyAdapter
+        }
+        assertTrue(adapter != newAdapter)
```

Comment:
Can you confirm that this test works if the adapter is not correct anyways ? I'm not sure about the != operator, it might just return false because the pointers are not the same (Like in Java). In that case maybe comparing the size of the datasets could be helpful

Reply:
yes but if the data changes but the size remain the same then the test will fail

Reply:
Ok I see, then maybe testing the content itself ?